LunrCoreLmdb/LmdbIndex.cs

```diff
@@ -0,0 +1,329 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using LightningDB;
+using Lunr;
+
+namespace LunrCoreLmdb
+{
+    public sealed class LmdbIndex
+    {
+        public Lazy<LightningEnvironment> Env { get; }
+
+        public string Path { get; }
+
+        public LmdbIndex(string path)
+        {
+            Env = new Lazy<LightningEnvironment>(() =>
+            {
+                var config = new EnvironmentConfiguration
+                {
+                    MaxDatabases = DefaultMaxDatabases,
+                    MaxReaders = DefaultMaxReaders,
+                    MapSize = DefaultMapSize
+                };
+                var environment = new LightningEnvironment(path, config);
+                environment.Open();
+                CreateIfNotExists(environment);
+                return environment;
+            });
+            Path = Env.Value.Path;
+        }
+
+		#region Fields 
+
+        public bool AddField(string field, CancellationToken cancellationToken = default)
+        {
+			cancellationToken.ThrowIfCancellationRequested();
+
+            using var tx = Env.Value.BeginTransaction(TransactionBeginFlags.None);
+            using var db = tx.OpenDatabase(configuration: Config);
+
+            tx.Put(db, KeyBuilder.BuildFieldKey(field), Encoding.UTF8.GetBytes(field), PutOptions.NoDuplicateData);
+            return tx.Commit() == MDBResultCode.Success;
+        }
+
+        public bool RemoveField(string field, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            using var tx = Env.Value.BeginTransaction(TransactionBeginFlags.None);
+            using var db = tx.OpenDatabase(configuration: Config);
+
+            tx.Delete(db, KeyBuilder.BuildFieldKey(field));
+            return tx.Commit() == MDBResultCode.Success;
+        }
+
+        public IEnumerable<string> GetFields(CancellationToken cancellationToken)
+        {
+            using var tx = Env.Value.BeginTransaction(TransactionBeginFlags.ReadOnly);
+            using var db = tx.OpenDatabase(configuration: Config);
+            using var cursor = tx.CreateCursor(db);
+
+            var sr = cursor.SetRange(KeyBuilder.BuildAllFieldsKey());
+            if (sr != MDBResultCode.Success)
+                yield break;
+
+            var (r, _, v) = cursor.GetCurrent();
+			
+            while (r == MDBResultCode.Success && !cancellationToken.IsCancellationRequested)
+            {
+                var buffer = v.AsSpan().ToArray();
+                yield return Encoding.UTF8.GetString(buffer);
```

Comment:
Can't we use the overload of `GetString` that takes a span rather than allocate an intermediary array?

Reply:
Yes, but then it would not be able to compile in `netstandard2.0`, which is what `LunrCore` itself is. I think there are workarounds using `System.Memory`. Is that what you want to do?

Reply:
Good point about netstandard 2.0. Didn't think it through.